project/src/com/google/daggerquery/plugin/GraphConverter.java

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.google.daggerquery.plugin;
+
+import com.google.common.graph.Network;
+import com.google.daggerquery.protobuf.autogen.BindingGraphProto;
+import com.google.daggerquery.protobuf.autogen.BindingGraphProto.BindingGraph.ListWithDependencies;
+import com.google.daggerquery.protobuf.autogen.DependencyProto.Dependency;
+import dagger.model.Binding;
+import dagger.model.BindingGraph;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class GraphConverter<NodeT, EdgeT> {
```

Comment:
Could you add a javadoc comment on the class, maybe specifying what NodeT and EdgeT are?

Reply:
Great idea, thanks, I've added it ✔️